kong/core/router.lua

```diff
@@ -600,8 +601,22 @@ function _M.new(apis)
     end
 
 
-    if api_t.upstream.path then
-      new_uri = api_t.upstream.path .. new_uri
+    local upstream_path = api_t.upstream.path
+    if upstream_path then
+      if new_uri == "/" then
+        new_uri = upstream_path
+      else
```

Comment:
I can do that, is it so that we are basically using new style on new code (or changes) even if the rest of the file is in other style?